Assets/Mirror/Components/NetworkSceneChecker.cs

```diff
@@ -0,0 +1,104 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.SceneManagement;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Component that controls visibility of networked objects between scenes.
+    /// <para>Any object with this component on it will only be visible to other objects in the same scene</para>
+    /// <para>This would be used when the server has multiple additive subscenes loaded to isolate players to their respective subscenes</para>
+    /// </summary>
+    [AddComponentMenu("Network/NetworkSceneChecker")]
+    [RequireComponent(typeof(NetworkIdentity))]
+    [HelpURL("https://mirror-networking.com/docs/Components/NetworkSceneChecker.html")]
+    public class NetworkSceneChecker : NetworkBehaviour
+    {
+        /// <summary>
+        /// How often (in seconds) that this object should update the list of observers that can see it.
+        /// </summary>
+        [Tooltip("How often (in seconds) that this object should check for scene change. Set to zero to disable.")]
+        public float updateInterval = 1;
+
+        /// <summary>
+        /// Flag to force this object to be hidden from all observers.
+        /// <para>If this object is a player object, it will not be hidden for that client.</para>
+        /// </summary>
+        [Tooltip("Enable to force this object to be hidden from all observers.")]
+        public bool forceHidden;
+
+        float lastUpdateTime;
+
+        Scene currentScene;
+
+        void Update()
+        {
+            if (!NetworkServer.active || updateInterval == 0)
+                return;
+
+            if (currentScene == null)
+                currentScene = gameObject.scene;
+
+            if (Time.time - lastUpdateTime > updateInterval)
```

Comment:
Why not InvokeRepeating instead?   And yes,  I am aware we do this in NetworkProximityChecker

Reply:
InvokeRepeating is a bit slow. this is probably better for MMO scale

Reply:
I'll do whatever Network Proximity Checker does...if you change that one I'll make this the same.

Reply:
I'll do whatever Network Proximity Checker does...if you change that one I'll make this the same.